bentoml/_internal/runner/runner.py

```diff
@@ -78,19 +77,30 @@ class BatchOptions:
 
 
 class _BaseRunner:
-    name: str
-    resource_quota: "ResourceQuota"
-    batch_options: "BatchOptions"
+    EXIST_NAMES: t.Set[str] = set()
 
+    @final
     def __init__(
         self,
-        name: str,
+        display_name: str,
         resource_quota: t.Optional[t.Dict[str, t.Any]] = None,
         batch_options: t.Optional[t.Dict[str, t.Any]] = None,
     ):
+        # probe an unique name
+        i = 0
+        while True:
+            name = display_name if i == 0 else f"{display_name}-{i}"
+            if name not in self.EXIST_NAMES:
+                self.EXIST_NAMES.add(name)
+                break
+            else:
+                i += 1
```

Comment:
Maybe we can apply the object id here instead of in the framework runner class?
```suggestion
        name = f"{display_name}-{id(self)}"
```

Reply:
This is still not ideal tho, in an idea case we would have identical runner name even if we restart the entire API server or run it from a different machine, if the runner is loaded from the same model with the same config